### ISSUE TITLE
Consistent use of term "tag" (re open)

### ISSUE THREAD

**Issue by laurencelundblade** [ietf-rats-wg/draft-ietf-rats-msg-wrap#148]
This is to reopen #133 

The double meaning of CMW CBOR Tag wasn't fixed (3.2 vs 10.3)

Please find an alternate to "CMW CBOR Tag". It sounds too much like what is in 10.3 and is out of line with common use.

The grammar of the first sentence isn't right.

Note that I was not asked to sign off on the issue even though I was the one that opened it.

**Comment by thomas-fossati** [comment]
> The double meaning of CMW CBOR Tag wasn't fixed (3.2 vs 10.3)
> 
> Please find an alternate to "CMW CBOR Tag". It sounds too much like what is in 10.3 and is out of line with common use.

Would "CMW Tag" work?  If not, can you suggest an alternative?

> The grammar of the first sentence isn't right.

The context is a bit vague :-)

**Comment by thomas-fossati** [comment]
@laurencelundblade, see 👆.  Would that work for you?  Do you have a better proposal?

**Comment by nedmsmith** [comment]
> "CMW TN"

I prefer "CMW TN Tag" to distinguish it as a "tag" since other text uses either record, collection, or tag.

**Comment by thomas-fossati** [comment]
> > "CMW TN"
> 
> I prefer "CMW TN Tag" to distinguish it as a "tag" since other text uses either record, collection, or tag.

I am fine with it.

**Comment by laurencelundblade** [comment]
"CoSWID tag", "CWT CBOR Tag" or "UCCS tag" and "CMW tag" (CMW section 10.3) are all things that have an assigned tag number and defined tag content. This here is not that at all.

So how about "CMW Tag Receptacle" or "CMW Tag Socket"? "CMW Tagged Item" is maybe another. "Tagged" distinguishes from "Tag".

The phrase to name this should not end in "tag" to distinguish from "CMW tag" and the like.

**Comment by thomas-fossati** [comment]
> "CoSWID tag", "CWT CBOR Tag" or "UCCS tag" and "CMW tag" (CMW section 10.3) are all things that have an assigned tag number and defined tag content. This here is not that at all.
> 
> So how about "CMW Tag Receptacle" or "CMW Tag Socket"? "CMW Tagged Item" is maybe another. "Tagged" distinguishes from "Tag".
> 
> The phrase to name this should not end in "tag" to distinguish from "CMW tag" and the like.

I am fine with "CMW Tagged Item".

**Comment by deeglaze** [comment]
I don’t see “cmw tag” in the document. There’s a CBOR tag (major type 6). There is a use of the same tag mechanism for conveying content format using the TN transformation. The other documents about CoSWID being a “tag” is inherited from the ISO wording for SWIDs about the metadata tag (like on clothing) to provide more information about the software artifact.

What construct are you suggesting to refer to as a cmw tagged item?

**Comment by thomas-fossati** [comment]
> I don’t see “cmw tag” in the document. There’s a CBOR tag (major type 6). There is a use of the same tag mechanism for conveying content format using the TN transformation. The other documents about CoSWID being a “tag” is inherited from the ISO wording for SWIDs about the metadata tag (like on clothing) to provide more information about the software artifact.
> 
> What construct are you suggesting to refer to as a cmw tagged item?

It's "CMW CBOR Tag", or sometimes just "CBOR Tag".  See https://github.com/ietf-rats-wg/draft-ietf-rats-msg-wrap/pull/172/files

**Comment by henkbirkholz** [comment]
> "CoSWID tag", "CWT CBOR Tag" or "UCCS tag" and "CMW tag" (CMW section 10.3) are all things that have an assigned tag number and defined tag content.

This list make it sound like they are the same things, but I would assume they are not.
* a CoSWID tag is a CBOR data item (no CBOR tag necessarily)
* a CBOR tagged CoSWID tag is a tagged CBOR data item with tag number: #6.1398229316

In case of CoSWID the "tag" terminology inherited from ISO and conflicts with the "tag" terminology from CBOR. That could not be changed and had to be dealt with. That is why awesome text, such as "CBOR tagged CoSWID tag" come to be. Not great, but at least also not impossible to grasp.

* a CWT CBOR Tag is a tagged COSE Envelope (a CBOR item) with the payload that is a CBOR Claims Set and with tag number: #6.61
* a UCCS tag is a tagged CWT Claims Set (a CBOR item) with tag number: #6.CPA601 (to be registered)

In summary, some times tags refer to data items (in cases of CoSWID or CoRIM) and sometimes tags refer to... well CBOR tags (as with CWT or UCCS examples above=

I'd say it is strongly implied that a "CBOR tagged item" is a "CBOR tagged data item" and not a "CBOR tag". Please see this table as an example https://www.ietf.org/archive/id/draft-ietf-rats-uccs-12.html#name-values-for-tags for the use of data item in the context of CBOR tags.

As a result, I would recommend not using the term item (or data item) in this context as reflected by this suggestion to #177: https://github.com/ietf-rats-wg/draft-ietf-rats-msg-wrap/pull/177/files#r1956236539

**Comment by laurencelundblade** [comment]
The distinction I'm after here is plug vs socket. I'm trying to use that distinction to come up with clarifying names for the draft.

"CWT CBOR Tag" or "UCCS tag" and "CMW tag"  are all plug-type things. (Left off CoSWID because it is ...) The definitions of all of these give a specific tag number and describe the structure of the tag content. The term "tag" is commonly used to mean a plug-type thing.

This is not the definition of a tag number and corresponding data structure, but a place where you plug in something with a tag number and data structure.

I don't mind "CBOR tagged blob" or "CBOR tagged message" or similar if you want to avoid item.

**Comment by nedmsmith** [comment]
Tag can also mean tag-value-pair (TLV) but more commonly is referred to as key-value.
It's convenient to be able to refer to the "CBOR tag" when talking about the parts of a CBOR tagged message. Aren't all of the CMW use cases aimed at content tagging for parser efficiency? 

If trying to describe key-value consructs like adding a label to a content value, then call it a key / key label rather than a tag.

**Comment by laurencelundblade** [comment]
@nedmsmith   Nothing to do with key-value here (that's CBOR maps, JSON objects). 

Here a tag is an alternate form of content type (MIME type) or CoAP Format ID. Tags are THE type extension mechanism in CBOR. Much use is to create data formats like dates and big numbers from basic CBOR types like ints and strings.

It's become a bit of a CBOR convention to use tags to indicate message and content types too. The problem, is that tags are CBOR only. There's no JSON equivalent. So in a mixed format world like RATS, they are not useful (The EAT submods design is poor because it relied on tags, not content type). 

In CMW, my understanding is that tags are here only for backwards compatibility with some early CMW implementations. There's general agreement that we should be using CoAP Format ID for the different types of RATS messages. 

Personally, I'd be fine with removing them altogether from CMW, but that's another discussion. They've been one of the most difficult things to design a good API for in my CBOR, COSE and EAT implementation. They don't make code smaller.

CoAP Content Format IDs are small integers so they're very efficient too.

**Comment by laurencelundblade** [comment]
Yes, I think that works.

**Comment by nedmsmith** [comment]
WFM

**Comment by thomas-fossati** [comment]
Earing no pushback, I'll make a PR.
